Tests/Common/ZipFileChecker.cs

```diff
@@ -90,6 +88,9 @@ public void AssertZipContainsOnlyExpectedFiles(params string[] expectedRelativeP
             string[] allFilesInZip = Directory.GetFiles(unzippedDirectoryPath, "*.*", SearchOption.AllDirectories);
             allFilesInZip.Length.Should().Be(expectedRelativePaths.Length, "Zip contains more files than expected");
         }
+        
+        public void Dispose() =>
+            archiveFile?.Dispose();
```

Comment:
`OpenRead` should not return `null`. It woudl throw instead and class would not be constructed
```suggestion
            archiveFile.Dispose();
```